### PR TITLE
aws: Adapt aws provider code to use the common provider framework

### PR DESCRIPTION
**Pull request by bpradipt**
Fixes:#579

Signed-off-by: Pradipta Banerjee <pradipta.banerjee@gmail.com>

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,9 +8,11 @@ umoci
...
 
 # Binaries
-agent-protocol-forwarder
-cloud-api-adaptor
-kata-agent
+# The leading `/` avoids recursive matching thereby ensuring
+# the code files under cmd/cloud-api-adaptor are not ignored
+/agent-protocol-forwarder
+/cloud-api-adaptor
+/kata-agent
```
Comment (yoheiueda) [inline .gitignore:15]: Do we need to change these lines as suggested at https://github.com/confidential-containers/cloud-api-adaptor/issues/578#issuecomment-1413422859?

File: pkg/adaptor/hypervisor/registry/aws.go [hunk 2]
```diff
@@ -1,13 +1,13 @@
+//go:build aws
 // +build aws
```
Comment (snir911) [inline pkg/adaptor/hypervisor/registry/aws.go:2]: nit: while here maybe remove the old style tag

File: .gitignore (hunk 1 above)
Comment (bpradipt) [reply .gitignore]: Added

File: pkg/adaptor/hypervisor/registry/aws.go (hunk 2 above)
Comment (bpradipt) [reply pkg/adaptor/hypervisor/registry/aws.go]: Fixed

**Review by yoheiueda — APPROVED** [review]
LGTM

**Review by snir911 — APPROVED** [review]
LGTM Thanks !

**Pull request merged**